commit 275e5c44679dcaed07007a8fbc634721afc7020d
Author: Nicholas Piggin <npiggin@gmail.com>
Date:   Fri Nov 10 04:27:38 2017 +1100

    powerpc/64s/hash: Allow MAP_FIXED allocations to cross 128TB boundary
    
    commit 35602f82d0c765f991420e319c8d3a596c921eb8 upstream.
    
    While mapping hints with a length that cross 128TB are disallowed,
    MAP_FIXED allocations that cross 128TB are allowed. These are failing
    on hash (on radix they succeed). Add an additional case for fixed
    mappings to expand the addr_limit when crossing 128TB.
    
    Fixes: f4ea6dcb08ea ("powerpc/mm: Enable mappings above 128TB")
    Signed-off-by: Nicholas Piggin <npiggin@gmail.com>
    Reviewed-by: Aneesh Kumar K.V <aneesh.kumar@linux.vnet.ibm.com>
    Signed-off-by: Michael Ellerman <mpe@ellerman.id.au>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

diff --git a/arch/powerpc/mm/slice.c b/arch/powerpc/mm/slice.c
index 3889201b560c..a4f93699194b 100644
--- a/arch/powerpc/mm/slice.c
+++ b/arch/powerpc/mm/slice.c
@@ -418,7 +418,7 @@ unsigned long slice_get_unmapped_area(unsigned long addr, unsigned long len,
 	unsigned long high_limit;
 
 	high_limit = DEFAULT_MAP_WINDOW;
-	if (addr >= high_limit)
+	if (addr >= high_limit || (fixed && (addr + len > high_limit)))
 		high_limit = TASK_SIZE;
 
 	if (len > high_limit)
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1170_275e5c44679dcaed07007a8fbc634721afc7020d.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1170_275e5c44679dcaed07007a8fbc634721afc7020d.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
